File: 프로그래머스/1/12926. 시저 암호/시저 암호.java
import java.util.*;
class Solution {
    public String solution(String s, int n) {
        String answer = "";

        StringBuilder sb = new StringBuilder();
        
        for(int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            // 공백
            if(c == ' '){
                sb.append(' ');
                continue;
            }
            // 소문자
            if(c >= 97){
                if(c + n > 122){
                    sb.append( (char) ((c + n) - 122 + 96) );
                }else {
                    sb.append( (char) (c + n) );
                }
                continue;
            }
            // 대문자
            if(c >= 65){
                if(c + n > 90){
                    sb.append( (char) ((c + n) - 90 + 64) );
                }else {
                    sb.append( (char) (c + n) );
                }
                continue;
            }
            
        }
        
        answer = sb.toString();
        
        return answer;
    }
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.py
def solution(s, n):
    answer = ''
    
    for c in s:
        if c == ' ':
            answer += ' '
        elif c.islower():
            answer += chr((ord(c) - 97 + n) % 26 + 97) # a
        elif c.isupper():
            answer += chr((ord(c) - 65 + n) % 26 + 65) # A
    
    return answer